main.cpp

```diff
@@ -38,6 +40,24 @@ void LogSettings() {
 
 } // namespace
 
+const CommandLineOptions& getCommandLineOptions() {
```

Comment:
This is kinda ugly as essentially a global variable. How are defaults and such handled? Can we just pass in the referenced values in arguments in the constructor?

Reply:
I was thinking in case of restarts (from updater-update or Windows requesting elevation), I might want to forward arguments to the next instance, for example to make it append to the same log file. But I don't think I'll implement that now... so this is lacking in motivation and I'll remove it.

Reply:
I got rid of the global accessor; was a bit gruesome with 3 classes to pass through. It was also impossible to require an argument in the QmlDownloader constructor because registering it as a QML type requires a 0-argument constructor, even if it is never constructed from QML. So I set it after construction